lib/ts/recipe/accountlinking/recipe.ts

```diff
@@ -17,82 +17,636 @@ import error from "../../error";
 import { BaseRequest, BaseResponse } from "../../framework";
 import normalisedURLPath from "../../normalisedURLPath";
 import RecipeModule from "../../recipeModule";
-import SuperTokens from "../..";
-import { AccountInfoWithRecipeId, APIHandled, HTTPMethod } from "../../types";
+import SuperTokensModule from "../../supertokens";
+import type { APIHandled, HTTPMethod, NormalisedAppinfo, RecipeListFunction, User } from "../../types";
 import { SessionContainer } from "../session";
-import { AccountInfoAndEmailWithRecipeId } from "./types";
-import { User } from "../../types";
+import type {
+    TypeNormalisedInput,
+    RecipeInterface,
+    TypeInput,
+    AccountInfoAndEmailWithRecipeId,
+    AccountInfoWithRecipeId,
+} from "./types";
+import { getUserForRecipeId, validateAndNormaliseUserInput } from "./utils";
+import OverrideableBuilder from "supertokens-js-override";
+import RecipeImplementation from "./recipeImplementation";
+import { Querier } from "../../querier";
+import SuperTokensError from "../../error";
 
-export default class AccountLinkingRecipe extends RecipeModule {
-    // recipeInterfaceImpl: RecipeInterface; TODO
+export default class Recipe extends RecipeModule {
+    private static instance: Recipe | undefined = undefined;
+
+    static RECIPE_ID = "accountlinking";
+
+    config: TypeNormalisedInput;
+
+    recipeInterfaceImpl: RecipeInterface;
+
+    constructor(recipeId: string, appInfo: NormalisedAppinfo, config: TypeInput, _recipes: {}, _ingredients: {}) {
+        super(recipeId, appInfo);
+        this.config = validateAndNormaliseUserInput(appInfo, config);
+
+        {
+            let builder = new OverrideableBuilder(
+                RecipeImplementation(Querier.getNewInstanceOrThrowError(recipeId), this.config)
+            );
+            this.recipeInterfaceImpl = builder.override(this.config.override.functions).build();
+        }
+    }
+
+    static init(config: TypeInput): RecipeListFunction {
+        return (appInfo) => {
+            if (Recipe.instance === undefined) {
+                Recipe.instance = new Recipe(
+                    Recipe.RECIPE_ID,
+                    appInfo,
+                    config,
+                    {},
+                    {
+                        emailDelivery: undefined,
+                    }
+                );
+                return Recipe.instance;
+            } else {
+                throw new Error("AccountLinking recipe has already been initialised. Please check your code for bugs.");
+            }
+        };
+    }
+
+    static getInstanceOrThrowError(): Recipe {
+        if (Recipe.instance === undefined) {
+            Recipe.init({})(
+                SuperTokensModule.getInstanceOrThrowError().appInfo,
+                SuperTokensModule.getInstanceOrThrowError().isInServerlessEnv
+            );
```

Comment:
this needs to be removed right? Since you are auto initialising it in supertokens.ts if the user has not initialised it already.

Reply:
done